add example of use of SetupMTCPathIK

marioprats opened the pull request:
Example for the Behavior added in https://github.com/PickNikRobotics/moveit_pro/pull/13712

rlpratt12 reviewed (commented):
Added a couple more test/demo Objectives. Not sure how detailed/complex we want this example to be, or if it makes sense to expose this, but I think the comparison showing `PlanCartesianPath` failing and `SetupMTCPathIK` succeeding in the complex case is pretty helpful.

marioprats commented:
See the comment in [my other PR](https://github.com/PickNikRobotics/moveit_pro/pull/13712). Since the MTC IK is still non-deterministic, I think it's going to be hard to have a difficult planning problem consistently succeeeding. I would keep it simple for now. Even if that same Objective can be planned with `PlanCartesianPath`, it's still an example of how to do it with MTC, and by keeping it simple it's hopefully less flaky.

marioprats commented:
> Played with this a little more, and I do think we just really need to make it more robust, so I added your suggested follow-up task: [PickNikRobotics/moveit_pro#13846](https://github.com/PickNikRobotics/moveit_pro/issues/13846)

Thanks. I created [this other example](https://github.com/PickNikRobotics/moveit_pro_example_ws/pull/394) (the second one, in factory_sim), which uses `SetupMTCPathIK` in a more complex setup. I simplified it a bit because otherwise I also started seeing the flakiness because on non-determinism. But it still shows a more real use case, so I think we can close this one, wdyt?
